opm/material/fluidsystems/blackoilpvt/DryHumidGasPvt.hpp

```diff
@@ -643,7 +643,8 @@ class DryHumidGasPvt
         //Prune duplicate Rv values (can occur, and will cause problems in further interpolation)
         auto x_coord_comparator = [](const Pair& a, const Pair& b) { return a.first == b.first; };
         auto last = std::unique(pSatSamplePoints.begin(), pSatSamplePoints.end(), x_coord_comparator);
-        pSatSamplePoints.erase(last, pSatSamplePoints.end());
+        if (pSatSamplePoints.begin() + 1 != last) // only remove them if there are more than two points
```

Comment:
I think I'd prefer to phrase this condition in terms of [`std::distance()`](https://en.cppreference.com/w/cpp/iterator/distance) instead.  The expression is **probably** well-defined behaviour as written, but `std::distance()` has fewer pitfalls:
```C++
if (std::distance(x.begin(), last) > 1) {
    // Prune duplicate elements if more than two unique elements in sequence
}
```

Reply:
Thanks.